tests/ferret/core_test.go

```diff
@@ -153,6 +153,8 @@ func TestCore(t *testing.T) {
 		testCases := []struct {
 			name string // TODO move to map key
 			q    bson.D
+			o    *options.FindOptions
+			v    bson.D
```

Comment:
I think it is worth mentioning what is this, why we need it in a comment.

And then I think we should check that only one of (v, IDs, err) is set (and mention that in the comment too)